src/scope.cc

```diff
@@ -4,6 +4,35 @@
 namespace Kakoune
 {
 
+void Scope::add_linked(SharedScope& scope, Context& context) {
+    if (contains(m_linked, &scope)) {
+        throw runtime_error("scope already linked");
+    }
+    m_linked.emplace_back(&scope);
+    
+    m_hooks.add_linked(scope.hooks());
+    
+    m_hooks.run_hook(Hook::ScopeLinked, scope.get_name(), context);
```

Comment:
Should these hooks contain information about which scope the shared scope was linked into?
That would require `get_name()` to be implemented for all scopes, but that shouldn't be a big issue I think. I am not entirely sure on the use cases here however, and would probably need some feedback on that.